kamu-cli/src/commands/log_command.rs

```diff
@@ -132,21 +135,34 @@ trait MetadataRenderer {
 
 struct AsciiRenderer {
     id_to_name_lookup: BTreeMap<DatasetID, DatasetName>,
+    limit: usize,
 }
 
 impl AsciiRenderer {
-    fn new(id_to_name_lookup: BTreeMap<DatasetID, DatasetName>) -> Self {
-        Self { id_to_name_lookup }
+    fn new(id_to_name_lookup: BTreeMap<DatasetID, DatasetName>, limit: usize) -> Self {
+        Self {
+            id_to_name_lookup,
+            limit,
+        }
     }
 
     async fn render_blocks<'a, 'b>(
         &'a self,
         output: &mut impl Write,
-        mut blocks: DynMetadataStream<'b>,
+        blocks: DynMetadataStream<'b>,
     ) -> Result<(), CLIError> {
+        let mut blocks = blocks.take(self.limit);
+
+        // TODO: We buffer output per block as writing directly to minus::Pager seems to
+        // have a lot of overhead. In future we should improve its async paging support.
+        let mut buf = Vec::new();
```

Comment:
I probably could wrap the `output` with some sort of fixed-size buffered writer, but I didn't bother, thinking we'll revisit this anyway and implement proper async pagination.